src/plugins/intel_npu/tools/single-image-test/main.cpp

```diff
@@ -1855,6 +1858,33 @@ static int runSingleImageTest() {
             inputFilesForOneInfer.push_back(std::move(entireModelFiles));
         }
 
+        std::vector<std::string> refFilesPerCase;
+        using RefFilesPerInput = std::vector<std::string>;
+        using RefFilesForModelOutputs = std::vector<RefFilesPerInput>;
+        std::vector<RefFilesForModelOutputs> refFilesForOneInfer;
+
+        if (!FLAGS_ref_results.empty()) {
+            refFilesPerCase = splitStringList(FLAGS_ref_results, ';');
+            // Make sure that the number of reference files is the same as the number of input files
+            if (refFilesPerCase.size() != inputFilesPerCase.size()) {
+                std::cout << "The number of reference files is not equal to the number of input files."
+                    << "  # Reference Files: " << refFilesPerCase.size()
+                    << "  # Input Files: " << inputFilesPerCase.size() << std::endl;
+                return EXIT_FAILURE;
+            }
+
+            for (const auto& refResult : refFilesPerCase) {
+                std::vector<std::string> refFilesPerModel = splitStringList(refResult, ',');
```

Comment:
AFAIR, --network supports only a single model to be loaded. Please explain when we expect several models in a single SIT run.

Reply:
> I guess that it is for batched input scenario when a whole row looks like `CASE_0;CASE_1;etc` where `CASE_X` is a run of SIT and which contains `INPUT_0,INPUT_1,INPUT2,etc`, where `INPUT_0` encompasses data for single input in batch `BATCH_0|BATCH_1|etc` which makes sense for separation input data per model

This is correct, the separators are mainly for delimiting if there are multiple expected outputs for the model.

Reply:
I see, but in order to leverage that "multiple expected outputs for the model" you need to write a special logic to split  "sigle batched output" onto several non-batched outputs - in order to make it works. Without that logic implemented this "|" is confusing as it doesn't work properly